meta.nabl2.runtime/trans/nabl2/runtime/analysis/generation.str

```diff
@@ -31,7 +31,12 @@ rules // top-level functions that are called during analysis
   )
 
   nabl2--generate-constraint-file(|source,args) = ?term; nabl2--strict(
-    <nabl2--generate-constraint(nabl2--generate-constraint-default-rule|"default")> (source,term,args)
+    nabl2_constraints := <nabl2--generate-constraint(nabl2--generate-constraint-default-rule|"default")> (source,term,args)
+  ; flowspec_constraints :=
+      <nabl2--generate-constraint-placeholder
+       <+ nabl2--generate-constraint-default-cfg-rule
+       <+ !CTrue(<nabl2--get-ast-index> term)> (source,term,args)
```

Comment:
Can you factor out lines 35 to 38?

Also, it is better to name the variable *_constraint, since it is not a list.